### PR TITLE
Fix scope encoding in token requests

### LINKED ISSUE #186: Scopes are encoded incorrectly when passed as an array
The README suggests that for the client credentials access token request and the resource owner  credentials access token request, multiple scopes can be specified as an array. However, doing so causes the scopes to be encoded incorrectly in the request body.

For example, using the client credentials grant with the `application/x-www-form-urlencoded` request content-type:

```javascript
const credentials = {
  client: { id: 'client-id', secret: 'client-secret' },
  auth: { tokenHost: 'http://localhost:3000' },
};
const oauth2 = require('simple-oauth2').create(credentials);
oauth2.clientCredentials.getToken({ scope: ['scope-1', 'scope-2'] });
```

Produces a request with the body

```
scope=scope-1&scope=scope-2&grant_type=client_credentials
```

I'd instead expect the request body to contain the scopes as space-separated like so:

```
scope=scope-1%20scope-2&grant_type=client_credentials
```

...to be consistent with [RFC 6749 § 3.3][spec]

[spec]: https://tools.ietf.org/html/rfc6749#section-3.3

### PR DESCRIPTION
**Pull request by cjlarose**
Fixes #186

### CODE REVIEW

File: test/.eslintrc.json [hunk 1]
```diff
@@ -1,7 +1,8 @@
 {
   "rules": {
     "max-len": "off",
-    "object-curly-newline": "off"
+    "object-curly-newline": "off",
+    "func-names": "off"
```
Comment (cjlarose) [inline test/.eslintrc.json]: I modified a few tests to write shared values to `this` and removed arrow function literals to do this.

The use of arrow function literals is [explicitly discouraged][1] by mocha, but I'm happy to switch the tests back to using block-scoped variables and arrow literals if preferred. 

[1]: https://mochajs.org/#arrow-functions

**Comment by yurevichcv** [comment]
Hi, when this PR will be merged?

**Comment by cjlarose** [comment]
@jonathansamines Anything I can do to help move this PR along?

File: test/.eslintrc.json (hunk 1 above)
Comment (jonathansamines) [reply test/.eslintrc.json]: Hi @cjlarose Sorry for the delay, please revert the mocha test changes related to the arrow functions usage. I am aware of what mocha suggests and open to changes, however in order to make your change easier to merge right now, I need you to restore those changes.

Comment (cjlarose) [reply test/.eslintrc.json]: Will do! 👍

**Comment by jonathansamines** [comment]
Resolved by #200

**Pull request closed without merging**